controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -120,134 +160,384 @@ func (r *StorageConsumerReconciler) Reconcile(ctx context.Context, request recon
+			string(r.storageConsumer.UID),
+			"ocs-storagecluster-cephfs"),
+		))
+		r.subVolumeGroupName = fmt.Sprintf("cephfilesystemsubvolumegroup-%s", hex.EncodeToString(cephFsMd5Sum[:16]))
+	} else {
+		// remote client
+		r.radosnamespaceName = string(r.storageConsumer.UID)
+		r.subVolumeGroupName = string(r.storageConsumer.UID)
+	}
 
-	r.noobaaAccount = &nbv1.NooBaaAccount{}
-	r.noobaaAccount.Name = r.storageConsumer.Name
-	r.noobaaAccount.Namespace = r.storageConsumer.Namespace
+	return nil
 }
 
 func (r *StorageConsumerReconciler) reconcilePhases() (reconcile.Result, error) {
+	if r.storageConsumer.GetDeletionTimestamp().IsZero() {
+		r.storageConsumer.Status.State = v1alpha1.StorageConsumerStateConfiguring
+		r.storageConsumer.Status.CephResources = []*ocsv1alpha1.CephResourcesSpec{}
 
-	if !r.storageConsumer.Spec.Enable {
-		r.storageConsumer.Status.State = v1alpha1.StorageConsumerStateDisabled
-		return reconcile.Result{}, nil
+		if err := r.initPhase(); err != nil {
+			return reconcile.Result{}, err
+		}
+
+		if err := r.reconcileCephRadosNamespace(); err != nil {
+			return reconcile.Result{}, err
+		}
+
+		if err := r.reconcileCephFilesystemSubVolumeGroup(); err != nil {
+			return reconcile.Result{}, err
+		}
+
+		if err := r.reconcileNoobaaAccount(); err != nil {
+			return reconcile.Result{}, err
+		}
+
+		if err := r.consolidateStatus(); err != nil {
+			return reconcile.Result{}, err
+		}
+	} else {
+		r.storageConsumer.Status.State = v1alpha1.StorageConsumerStateDeleting
 	}
 
-	r.storageConsumer.Status.State = v1alpha1.StorageConsumerStateConfiguring
-	r.storageConsumer.Status.CephResources = []*ocsv1alpha1.CephResourcesSpec{}
+	return reconcile.Result{}, nil
+}
 
-	if r.storageConsumer.GetDeletionTimestamp().IsZero() {
+func (r *StorageConsumerReconciler) reconcileCephRadosNamespace() error {
+	blockPools := &rookCephv1.CephBlockPoolList{}
+	if err := r.List(r.ctx, blockPools, client.InNamespace(r.namespace)); err != nil {
+		return err
+	}
 
-		// A provider cluster already has a NooBaa system and does not require a NooBaa account
-		// to connect to a remote cluster, unlike client clusters.
-		// A NooBaa account only needs to be created if the storage consumer is for a client cluster.
-		clusterID := util.GetClusterID(r.ctx, r.Client, &r.Log)
-		if clusterID != "" && !strings.Contains(r.storageConsumer.Name, clusterID) {
-			if err := r.reconcileNoobaaAccount(); err != nil {
-				return reconcile.Result{}, err
+	// ensure for this consumer a rados namespace is created in every blockpool
+	for idx := range blockPools.Items {
+		bp := &blockPools.Items[idx]
+		if bp.Name == "builtin-mgr" {
+			continue
+		}
+
+		rns := &rookCephv1.CephBlockPoolRadosNamespace{}
+		rns.Name = func() string {
+			if r.isUpdatedProvider && strings.HasPrefix(bp.Name, r.storageCluster.Name) {
+				return r.radosnamespaceName
+			}
+			return fmt.Sprintf("%s-%s", bp.Name, r.radosnamespaceName)
+		}()
```

Comment:
a comment here would be good, wrt Day 1 pool having storageCluster name as prefix and Day 2 is user choice, anon func isn't required here as there is ntg much to calculate.